### ISSUE TITLE
No handling of nix evaluation failures

### ISSUE THREAD

**Issue by erikarvstedt** [shajra/direnv-nix-lorelei#2]
When nix evaluation fails, no failure information is displayed. Instead, `direnv-nix-lorelei` fails with an internal error.

#### Reproduce
```
tmpDir=/tmp/lorelei-bug
rm -rf $tmpDir
mkdir -p $tmpDir

# Set this to the direnv-nix-lorelei src
lorelei=$(nix-build --no-out-link /path/to/src -A direnv-nix-lorelei)

cat > $tmpDir/.envrc <<EOF
. $lorelei/share/direnv-nix-lorelei/nix-lorelei.sh
use_nix_gcrooted -a
EOF

cat > $tmpDir/shell.nix <<EOF
fail
EOF

(cd $tmpDir; direnv allow; direnv exec . true)
rm -rf $tmpDir
```
Output
```
direnv: loading /tmp/lorelei-bug/.envrc
direnv: not modified: /tmp/direnv-layouts/@tmp@lorelei-bug/delete_to_rebuild
direnv: hash check invalidated cache
direnv: rebuilding with Nix
/nix/store/z1qvlavy35wanw5k54fvvfffws5bvigj-coreutils-8.31/bin/ln: failed to create symbolic link '/tmp/direnv-layouts/@tmp@lorelei-bug/env-' -> '': No such file or directory
direnv: watching /tmp/direnv-layouts/@tmp@lorelei-bug/delete_to_rebuild
direnv: watching .envrc
direnv: watching /tmp/lorelei-bug/shell.nix
```

Proper error message propagation will be hard to handle in an efficient streaming manner with bash.
Maybe the best workaround would be to re-run `nix-build` when a failure is detected in the first run.
Alternatively, I'd be happy to see `nix-lorelei` rewritten in another, more adequate language.

**Comment by shajra** [comment]
> Alternatively, I'd be happy to see nix-lorelei rewritten in another, more adequate language.

I think that _could_ be done.  It seems a bit heavy for a script that's still kind of small.  I'm delegating pretty heavily to Lorri.  The parts of Lorri I can delegate more easily to are their Nix expressions and associated shell snippets.  What I can't delegate as easily to is some code written in Rust.  So if someone were to actually try to write Lorelei in a better language, I'd suggest Rust.  Though I don't know Rust really, and my go-to language for that kind of thing is Haskell.

I'm really glad you're using Lorelei, though.   So let's see if we can get shell to do what we want without incurring a ton of awkward code.  I'll leave this issue open and see if I can find some time to sort out a solution.  Feel free to ping me if you get a little antsy for a resolution.

**Comment by shajra** [comment]
Fixed by #3.
